### ISSUE TITLE
no-unused-variable: private ctor params shouldn't require `this.` when used in the ctor

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#1336]
### Bug Report

- __TSLint version__: `3.11.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: `gulp-tslint` via Visual Studio 2015

#### TypeScript code being linted

```ts
class Example {
  constructor(private thing1: string, private thing2: string){
    console.log(thing1);
    console.log(this.thing2);
  }
}
```

with `tslint.json` configuration:

```json
"no-unused-variable": [ true, "check-parameters" ],
```

#### Actual behavior

The private `thing1` variable is reported as unused, even though it is used without the `this.` prefix, which is allowed in a constructor.

#### Expected behavior

Used variables in the constructor should not require the `this.` prefix to be considered "used"

**Comment by JKillian** [comment]
`this.thing1` vs. `thing1` have different meanings - consider [the case below][0]:

```
class Foo {
	constructor(public foo = 20) {
		this.foo = 10;
		foo = 5;
	}
}

const f = new Foo();
alert(f.foo);  // will display 10
```

If a private parameter property of a constructor is never accessed accessed anywhere in the class as `this.myProperty`, it actually still is unused as a class member and I think the lint rule is succeeding. Does this line up with what you're seeing from TSLint?

[0]: http://www.typescriptlang.org/play/#src=class%20Foo%20%7B%0A%09constructor(public%20foo%20%3D%2020)%20%7B%0A%09%09this.foo%20%3D%2010%3B%0A%09%09foo%20%3D%205%3B%0A%09%7D%0A%7D%0A%0Aconst%20f%20%3D%20new%20Foo()%3B%0Aalert(f.foo)%3B%0A%0A

**Comment by chrismbarr** [comment]
I was not aware of that, I assumed they meant the same thing.  Thanks for the clarification!

**Comment by chrismbarr** [comment]
Now that I'm thinking about this more... should there be a rule to ban the improper use like this?
```ts
constructor(public foo: number) {
        this.foo = 10;
        foo = 5; //<-- violation
}
```
